test_scripts/API/VehicleData/WindowStatus/003_GENERIC_ERROR_in_case_HMI_respond_with_invalid_data.lua

```diff
@@ -0,0 +1,131 @@
+    { location = { col = 49, RoW = 49 },
+      state = common.getWindowStatusParams()[1].state
+    }
+  },
+  ["empty_state"] = { -- empty state parameter
+    { location = common.getWindowStatusParams()[1].location,
+      state = {}
+    }
+  },
+  ["invalidType_state"] = { -- invalid type for state parameter
+    { location = common.getWindowStatusParams()[1].location,
+      state = "string"
+    }
+  },
+  ["missing_state"] = { -- without state parameter
+    { location = common.getWindowStatusParams()[1].location }
+  },
+  ["invalidName_state"] = { -- invalid name for state parameter
+    { location = common.getWindowStatusParams()[1].location,
+      StaTe = { approximatePosition = 50, deviation = 50 }
+    }
+  },
+  ["invalidName_approximatePosition"] = { -- invalid name for approximatePosition parameter from WindowState structure
+    { location = common.getWindowStatusParams()[1].location,
+      state = { ApproximatePositioN = 50, deviation = 50 }
+    }
+  },
+  ["invalidName_deviation"] = { -- invalid name for deviation parameter from WindowState structure
+    { location = common.getWindowStatusParams()[1].location,
+      state = { approximatePosition = 50, DeviatioN = 50 }
+    }
+  },
+  ["missing_col"] = { -- without col parameter from Grid structure
+    { location = { row = 49, level = 49, colspan = 49, rowspan = 49, levelspan = 49 },
+      state = { approximatePosition = 50, deviation = 50 }
+    }
+  },
+  ["missing_row"] = { -- without row parameter from Grid structure
+    { location = { col = 49, level = 49, colspan = 49, rowspan = 49, levelspan = 49 },
+      state = { approximatePosition = 50, deviation = 50 }
+    }
+  }
+}
+
+local resultCode = { success = false, resultCode = "GENERIC_ERROR" }
+
+--[[ Scenario ]]
+common.Title("Preconditions")
+common.Step("Clean environment", common.preconditions)
+common.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+common.Step("Register App", common.registerApp)
+common.Step("Activate App", common.activateApp)
+
+common.Title("Test")
+for p in pairs(common.getWindowStatusParams()[1].location) do
+  common.Title("Check " .. p .. " parameter from Grid structure")
+  for k, v in pairs(invalidValue) do
+    common.Step("HMI sends GetVehicleData response with invalid " .. p .. "=" .. tostring(k),
+      common.getVehicleData, { common.getCustomData(p, "location", v), resultCode })
+  end
+end
+
+for p in pairs(common.getWindowStatusParams()[1].state) do
+  common.Title("Check " .. p .. " parameter from WindowState structure")
+  for k, v in pairs(invalidValue) do
+    common.Step("HMI sends GetVehicleData response with invalid " .. p .. "=" .. tostring(k),
+      common.getVehicleData, { common.getCustomData(p, "state", v), resultCode })
+  end
+end
+
+common.Title("Check for other parameters")
+for k, v in pairs(invalidParam) do
+  common.Step("HMI sends GetVehicleData response with " .. k, common.getVehicleData, { v, resultCode })
+end
+
```

Comment:
in this script should be covered cases with array size: out of min and max bound